keps/sig-node/4815-dra-partitionable-devices/README.md

```diff
@@ -0,0 +1,2140 @@
+	//
+	// +optional
+	ConsumesCapacityFrom []DeviceRef `json:"consumesCapacityFrom,omitempty"`
+
+	// Attributes defines the set of attributes for this device.
+	// The name of each attribute must be unique in that set.
+	//
+	// To ensure this uniqueness, attributes defined by the vendor
+	// must be listed without the driver name as domain prefix in
+	// their name. All others must be listed with their domain prefix.
+	//
+	// Conflicting attributes from those provided via mixins are
+	// overwritten by the ones provided here.
+	//
+	// The maximum number of attributes and capacities combined is 32.
+	//
+	// +optional
+	Attributes map[QualifiedName]DeviceAttribute `json:"attributes,omitempty"`
+
+	// Capacity defines the set of capacities for this device.
+	// The name of each capacity must be unique in that set.
+	//
+	// To ensure this uniqueness, capacities defined by the vendor
+	// must be listed without the driver name as domain prefix in
+	// their name. All others must be listed with their domain prefix.
+	//
+	// Conflicting capacities from those provided via mixins are
+	// overwritten by the ones provided here.
+	//
+	// The maximum number of attributes and capacities combined is 32.
+	//
+	// +optional
+	Capacity map[QualifiedName]DeviceCapacity `json:"capacity,omitempty"`
+}
+
+// DeviceMixinRef defines a reference to a device mixin.
+type DeviceMixinRef struct {
+	// Name refers to the name of a device mixin in the pool.
+	//
+	// +required
+	Name string `json:"name"`
+}
+
+// DeviceRef defines a reference to a device.
+type DeviceRef struct {
+	// Name refers to the name of a device in the pool.
+	//
+	// +required
+	Name string `json:"name"`
+}
+```
+
+As mentioned previously, the main features being added here are (1) the ability
+to include a set of mixins in a device definition, and (2) the ability to
+express that capacity from one device gets consumed by another device if/when
+the scheduler decides to allocate it.
+
+To simplify the conversation, we discuss each new feature separately, starting
+with "mixins" and the new `Includes` field, which allows a set of mixins to
+extend a device with common attributes and capacities.
+
+### Extending a device with as set of mixins
+
+A simple example the defines a set of mixins and includes them in the
+definition of 4 NVIDIA A100 GPUs can be seen below:
+
+```yaml
+deviceMixins:
+- name: system-attributes
+  composite:
+    attributes:
+      cudaDriverVersion:
+        version: 12.6.0
+      driverVersion:
+        version: 560.35.3
+- name: common-gpu-attributes
+  composite:
+    attributes:
+      type:
+        string: gpu
+      architecture:
+        string: Ampere
+      brand:
+        string: Nvidia
+      productName:
+        string: NVIDIA A100-SXM4-40GB
+      cudaComputeCapability:
+        version: 8.0.0
+- name: common-gpu-capacities
+  composite:
+    capacity:
+      memory:
+        quantity: 40Gi
+devices:
+- name: gpu-0
+  composite:
```

Comment:
Yeah, this goes back to the comment above...

> I'd love it if we could just add both of these new fields to the existing BasicDevice type (since they are strictly extensions), but as @pohly keeps pointing out (because I keep forgetting) we can't extend the API in this way because it breaks downgrades. An old scheduler will not recognize the new fields and attempt to allocate a device based solely on the fields that it DOES know about. For mixins / attributes this is not too bad because selection would simply fail if the scheduler was not able to see the additional attributes that a mixin provides. Not honoring consumesCapacityFrom properly would be unacceptable though.